cstar/roms/component.py

```diff
@@ -114,23 +97,10 @@ def __init__(
         self.base_model: "ROMSBaseModel" = base_model
         self.additional_code: AdditionalCode = additional_code
         self.input_datasets: List["InputDataset"] = input_datasets or []
-
-        # QUESTION: should all these attrs be passed in as a single "discretization" arg of type dict?
-        self.time_step: int = time_step
-        self.nx: Optional[int] = nx
-        self.ny: Optional[int] = ny
-        self.n_levels: Optional[int] = n_levels
-        self.n_procs_x: Optional[int] = n_procs_x
-        self.n_procs_y: Optional[int] = n_procs_y
+        self.discretization = discretization
         self.exe_path: Optional[Path] = None
 
-    @property
-    def n_procs_tot(self) -> Optional[int]:
-        """Total number of processors required by this ROMS configuration"""
-        if (self.n_procs_x is None) or (self.n_procs_y is None):
-            return None
-        else:
-            return self.n_procs_x * self.n_procs_y
+        assert isinstance(self.discretization, ROMSDiscretization)
```

Comment:
Technically you shouldn't use `assert` outside of tests because supposedly some python compilers compile them away as an optimization. Never actually had this happen to me though.

```suggestion
        if not isinstance(self.discretization, ROMSDiscretization):
            raise TypeError()
```

Reply:
I know, this is really nasty, was just to avoid the linter losing the thread. I don't see why your implementation wouldn't achieve the same thing though, I'll try it and see what pre-commit does

Reply:
I just deleted it, seems fine now whatever the issue was before